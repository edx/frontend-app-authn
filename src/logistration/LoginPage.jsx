import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getConfig } from '@edx/frontend-platform';
import { Button, Input, ValidationFormGroup } from '@edx/paragon';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFacebookF, faGoogle, faMicrosoft } from '@fortawesome/free-brands-svg-icons';
import { getQueryParameters } from '@edx/frontend-platform';

import { loginRequest } from './data/actions';
import { loginRequestSelector } from './data/selectors';
import { forgotPasswordResultSelector } from '../forgot-password';
import ConfirmationAlert from './ConfirmationAlert';
import LoginHelpLinks from './LoginHelpLinks';


const LoginRedirect = (props) => {
  let { success, redirectUrl } = props;
  if (success) {
    const lmsBaseUrl = getConfig().LMS_BASE_URL;
    if (!redirectUrl.includes(lmsBaseUrl)) {
      redirectUrl = lmsBaseUrl + redirectUrl;
    }
    window.location.href = redirectUrl;
  }
  return <></>;
};

LoginRedirect.defaultProps = {
  redirectUrl: '',
  success: false,
};

LoginRedirect.propTypes = {
  redirectUrl: PropTypes.string,
  success: PropTypes.bool,
};

class LoginPage extends React.Component {
  constructor(props, context) {
    super(props, context);

    this.state = {
      password: '',
      email: '',
      errors: {
        email: '',
        password: '',
      },
      emailValid: false,
      passwordValid: false,
      formValid: false,
    };
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const params = getQueryParameters();
    const payload = {
      email: this.state.email,
      password: this.state.password,
    };
    if (params.next) {
      payload['next'] = params.next;
    }
    if (params.course_id) {
      payload['course_id'] = params.course_id;
    }

    if (!this.state.formValid) {
      this.validateInput('email', payload.email);
      this.validateInput('password', payload.password);
      return;
    }

    this.props.loginRequest(payload);
  }

  validateInput(inputName, value) {
    let { emailValid, passwordValid } = this.state;
    const { errors } = this.state;

    switch (inputName) {
      case 'email':
        emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
        errors.email = emailValid ? '' : null;
        break;
      case 'password':
        passwordValid = value.length > 0;
        errors.password = passwordValid ? '' : null;
        break;
      default:
        break;
    }

    this.setState({
      errors,
      emailValid,
      passwordValid,
    }, this.validateForm);
  }

  handleOnChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
    this.validateInput(e.target.name, e.target.value);
  }

  validateForm() {
    const { emailValid, passwordValid } = this.state;
    this.setState({
      formValid: emailValid && passwordValid,
    });
  }

  render() {
    return (
      <>
        <LoginRedirect success={this.props.loginResult.success} redirectUrl={this.props.loginResult.redirectUrl} />
        <div className="d-flex justify-content-center logistration-container">
          <div className="d-flex flex-column" style={{ width: '400px' }}>
            {this.props.forgotPassword.status === 'complete' ? <ConfirmationAlert email={this.props.forgotPassword.email} /> : null}
            <div className="d-flex flex-row">
              <p>
                First time here?<a className="ml-1" href="/register">Create an Account.</a>
              </p>
            </div>
            <form className="m-0">
              <div className="form-group">
                <h3 className="text-center mt-3">Sign In</h3>

                <div className="d-flex flex-column align-items-start">
                  <ValidationFormGroup
                    for="email"
                    invalid={this.state.errors.email !== ''}
                    invalidMessage="The email address you've provided isn't formatted correctly."
                  >
                    <label htmlFor="loginEmail" className="h6 mr-1">Email</label>
                    <Input
                      name="email"
                      id="loginEmail"
                      type="email"
                      placeholder="username@domain.com"
                      value={this.state.email}
                      onChange={e => this.handleOnChange(e)}
                      style={{ width: '400px' }}
                    />
                  </ValidationFormGroup>
                </div>
                <p className="mb-4">The email address you used to register with edX.</p>
                <div className="d-flex flex-column align-items-start">
                  <ValidationFormGroup
                    for="password"
                    invalid={this.state.errors.password !== ''}
                    invalidMessage="Please enter your password."
                    className="mb-0"
                  >
                    <label htmlFor="loginPassword" className="h6 mr-1">Password</label>
                    <Input
                      name="password"
                      id="loginPassword"
                      type="password"
                      value={this.state.password}
                      onChange={e => this.handleOnChange(e)}
                      style={{ width: '400px' }}
                    />
                  </ValidationFormGroup>
                </div>
                <LoginHelpLinks page="login" />
              </div>
              <Button
                className="btn-primary submit"
                onClick={this.handleSubmit}
              >
                Sign in
              </Button>
            </form>
            <div className="section-heading-line mb-4">
              <h4>or sign in with</h4>
            </div>
            <div className="row text-center d-block mb-4">
              <button type="button" className="btn-social facebook"><FontAwesomeIcon className="mr-2" icon={faFacebookF} />Facebook</button>
              <button type="button" className="btn-social google"><FontAwesomeIcon className="mr-2" icon={faGoogle} />Google</button>
              <button type="button" className="btn-social microsoft"><FontAwesomeIcon className="mr-2" icon={faMicrosoft} />Microsoft</button>
            </div>
          </div>
        </div>
      </>
    );
  }
}

LoginPage.defaultProps = {
  loginResult: null,
  forgotPassword: null,
};

LoginPage.propTypes = {
  loginRequest: PropTypes.func.isRequired,
  loginResult: PropTypes.shape({
    redirectUrl: PropTypes.string,
    success: PropTypes.bool,
  }),
  forgotPassword: PropTypes.shape({
    email: PropTypes.string,
    status: PropTypes.string,
  }),
};

const mapStateToProps = state => {
  const forgotPassword = forgotPasswordResultSelector(state);
  const loginResult = loginRequestSelector(state);
  return { forgotPassword, loginResult };
};

export default connect(
  mapStateToProps,
  {
    loginRequest,
  },
)(LoginPage);
