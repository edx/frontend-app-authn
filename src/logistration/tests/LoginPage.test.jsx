import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';

import { getConfig } from '@edx/frontend-platform';
import { IntlProvider, injectIntl } from '@edx/frontend-platform/i18n';

import LoginPage from '../LoginPage';

const IntlLoginPage = injectIntl(LoginPage);
const mockStore = configureStore();

describe('LoginPage', () => {
  const initialState = {
    logistration: {
      forgotPassword: { status: null },
      loginResult: { success: false, redirectUrl: '' },
      response_error: null,
      thirdPartyAuthContext: {
        currentProvider: null,
        finishAuthUrl: null,
        providers: [],
      },
    },
  };

  let props = {};
  let store = {};

  const appleProvider = {
    id: 'oa2-apple-id',
    name: 'Apple',
    iconClass: null,
    iconImage: 'https://edx.devstack.lms/logo.png',
    loginUrl: '/auth/login/apple-id/?auth_entry=login&next=/dashboard',
  };

  const reduxWrapper = children => (
    <IntlProvider locale="en">
      <Provider store={store}>{children}</Provider>
    </IntlProvider>
  );

  beforeEach(() => {
    store = mockStore(initialState);
    props = {
      loginRequest: jest.fn(),
    };
  });

  it('should match default section snapshot', () => {
    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match forget password alert message snapshot', () => {
    props = {
      ...props,
      forgotPassword: { status: 'complete', email: 'test@example.com' },
    };
    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should match TPA provider snapshot', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        thirdPartyAuthContext: {
          providers: [appleProvider],
        },
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show error message on 400', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        loginError: 'Email or password is incorrect.',
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show error message on 400 on receiving link', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        loginError: 'To be on the safe side, you can reset your password <a href="/reset">here</a> before you try again.\n',
      },
    });

    const tree = renderer.create(reduxWrapper(<IntlLoginPage {...props} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should display login help button', () => {
    const root = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(root.find('button.field-link').text()).toEqual('Need help signing in?');
  });

  it('updates the error state for invalid email', () => {
    const errorState = { email: null, password: null };
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));

    loginPage.find('button.submit').simulate('click');
    expect(loginPage.find('LoginPage').state('errors')).toEqual(errorState);
  });

  it('updates the error state for invalid password', () => {
    const errorState = { email: '', password: null };
    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));

    loginPage.find('input#loginEmail').simulate('change', { target: { value: 'test@example.com', name: 'email' } });
    loginPage.find('button.submit').simulate('click');
    expect(loginPage.find('LoginPage').state('errors')).toEqual(errorState);
  });

  it('should match url after redirection', () => {
    const dasboardUrl = 'http://test.com/testing-dashboard/';
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        loginResult: {
          success: true,
          redirectUrl: dasboardUrl,
        },
      },
    });
    delete window.location;
    window.location = { href: '' };
    renderer.create(reduxWrapper(<IntlLoginPage {...props} />));
    expect(window.location.href).toBe(dasboardUrl);
  });

  it('should match url after TPA redirection', () => {
    const authCompleteUrl = '/auth/complete/google-oauth2/';
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        loginResult: {
          success: true,
          redirectUrl: '',
        },
        thirdPartyAuthContext: {
          ...initialState.logistration.thirdPartyAuthContext,
          finishAuthUrl: authCompleteUrl,
        },
      },
    });

    delete window.location;
    window.location = { href: '' };
    renderer.create(reduxWrapper(<IntlLoginPage {...props} />));
    expect(window.location.href).toBe(getConfig().LMS_BASE_URL + authCompleteUrl);
  });

  it('should call the componentDidMount lifecycle method', () => {
    const spy = jest.spyOn(LoginPage.WrappedComponent.prototype, 'componentDidMount');

    mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(spy).toHaveBeenCalled();
  });

  it('should redirect to social auth provider url', () => {
    const loginUrl = '/auth/login/apple-id/?auth_entry=login&next=/dashboard';
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        thirdPartyAuthContext: {
          providers: [{
            ...appleProvider,
            loginUrl,
          }],
        },
      },
    });

    delete window.location;
    window.location = { href: '' };

    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));

    loginPage.find('button#oa2-apple-id').simulate('click');
    expect(window.location.href).toBe(getConfig().LMS_BASE_URL + loginUrl);
  });

  it('should match third party auth alert', () => {
    store = mockStore({
      ...initialState,
      logistration: {
        ...initialState.logistration,
        thirdPartyAuthContext: {
          ...initialState.logistration.thirdPartyAuthContext,
          currentProvider: 'Apple',
          platformName: 'edX',
        },
      },
    });

    const expectedMessage = 'You have successfully signed into Apple, but your Apple account does not have a '
                            + 'linked edX account. To link your accounts, sign in now using your edX password.';

    const loginPage = mount(reduxWrapper(<IntlLoginPage {...props} />));
    expect(loginPage.find('#tpa-alert').find('span').text()).toEqual(expectedMessage);
  });
});
