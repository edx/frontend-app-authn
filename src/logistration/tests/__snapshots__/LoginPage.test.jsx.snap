// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should match TPA provider snapshot 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="d-flex flex-row"
    >
      <p>
        First time here?
        <a
          className="ml-1"
          href="/register"
        >
          Create an Account
          .
        </a>
      </p>
    </div>
    <h2
      className="text-left mt-2 mb-3"
    >
      Sign In
    </h2>
    <form
      className="m-0 mw-420"
    >
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginEmail"
        >
          Email
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginEmail"
          name="email"
          onChange={[Function]}
          placeholder="username@domain.com"
          type="email"
          value=""
        />
        <p
          className="mb-4"
        >
          The email address you used to register with edX.
        </p>
        <strong
          className="invalid-feedback"
          id="email-invalid-feedback"
        >
          The email address you've provided isn't formatted correctly.
        </strong>
      </div>
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginPassword"
        >
          Password
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginPassword"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <strong
          className="invalid-feedback"
          id="password-invalid-feedback"
        >
          Please enter your password.
        </strong>
      </div>
      <button
        className="mt-1 field-link"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
          data-icon="caret-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Need help signing in?
      </button>
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      >
        <div
          style={
            Object {
              "padding": ".1px 0",
            }
          }
        />
      </div>
      <a
        className="field-link mt-0 mb-3"
        href="http://localhost:18000/enterprise/login"
        onClick={[Function]}
        target="_self"
      >
        Sign in with your company or school
      </a>
      <button
        aria-disabled={false}
        aria-live="assertive"
        className="pgn__stateful-btn pgn__stateful-btn-state-default btn-primary submit btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <span
          className="d-flex align-items-center justify-content-center"
        >
          Sign in
        </span>
      </button>
    </form>
    <div
      className="section-heading-line mb-4"
    >
      <h4>
        or sign in with
      </h4>
    </div>
    <div
      className="row tpa-container"
    >
      <button
        className="btn-social btn-oa2-apple-id"
        data-provider-url="/auth/login/apple-id/?auth_entry=login&next=/dashboard"
        id="oa2-apple-id"
        onClick={[Function]}
        type="button"
      >
        <div
          aria-hidden="true"
        >
          <img
            alt="icon Apple"
            className="icon-image"
            src="https://edx.devstack.lms/logo.png"
          />
        </div>
        <span
          aria-hidden="true"
          className="pl-2"
        >
          Apple
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginPage should match default section snapshot 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="d-flex flex-row"
    >
      <p>
        First time here?
        <a
          className="ml-1"
          href="/register"
        >
          Create an Account
          .
        </a>
      </p>
    </div>
    <h2
      className="text-left mt-2 mb-3"
    >
      Sign In
    </h2>
    <form
      className="m-0 mw-420"
    >
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginEmail"
        >
          Email
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginEmail"
          name="email"
          onChange={[Function]}
          placeholder="username@domain.com"
          type="email"
          value=""
        />
        <p
          className="mb-4"
        >
          The email address you used to register with edX.
        </p>
        <strong
          className="invalid-feedback"
          id="email-invalid-feedback"
        >
          The email address you've provided isn't formatted correctly.
        </strong>
      </div>
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginPassword"
        >
          Password
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginPassword"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <strong
          className="invalid-feedback"
          id="password-invalid-feedback"
        >
          Please enter your password.
        </strong>
      </div>
      <button
        className="mt-1 field-link"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
          data-icon="caret-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Need help signing in?
      </button>
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      >
        <div
          style={
            Object {
              "padding": ".1px 0",
            }
          }
        />
      </div>
      <a
        className="field-link mt-0 mb-3"
        href="http://localhost:18000/enterprise/login"
        onClick={[Function]}
        target="_self"
      >
        Sign in with your company or school
      </a>
      <button
        aria-disabled={false}
        aria-live="assertive"
        className="pgn__stateful-btn pgn__stateful-btn-state-default btn-primary submit btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <span
          className="d-flex align-items-center justify-content-center"
        >
          Sign in
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`LoginPage should match forget password alert message snapshot 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="d-flex flex-row"
    >
      <p>
        First time here?
        <a
          className="ml-1"
          href="/register"
        >
          Create an Account
          .
        </a>
      </p>
    </div>
    <h2
      className="text-left mt-2 mb-3"
    >
      Sign In
    </h2>
    <form
      className="m-0 mw-420"
    >
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginEmail"
        >
          Email
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginEmail"
          name="email"
          onChange={[Function]}
          placeholder="username@domain.com"
          type="email"
          value=""
        />
        <p
          className="mb-4"
        >
          The email address you used to register with edX.
        </p>
        <strong
          className="invalid-feedback"
          id="email-invalid-feedback"
        >
          The email address you've provided isn't formatted correctly.
        </strong>
      </div>
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginPassword"
        >
          Password
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginPassword"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <strong
          className="invalid-feedback"
          id="password-invalid-feedback"
        >
          Please enter your password.
        </strong>
      </div>
      <button
        className="mt-1 field-link"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
          data-icon="caret-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Need help signing in?
      </button>
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      >
        <div
          style={
            Object {
              "padding": ".1px 0",
            }
          }
        />
      </div>
      <a
        className="field-link mt-0 mb-3"
        href="http://localhost:18000/enterprise/login"
        onClick={[Function]}
        target="_self"
      >
        Sign in with your company or school
      </a>
      <button
        aria-disabled={false}
        aria-live="assertive"
        className="pgn__stateful-btn pgn__stateful-btn-state-default btn-primary submit btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <span
          className="d-flex align-items-center justify-content-center"
        >
          Sign in
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`LoginPage should match pending button state snapshot 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="d-flex flex-row"
    >
      <p>
        First time here?
        <a
          className="ml-1"
          href="/register"
        >
          Create an Account
          .
        </a>
      </p>
    </div>
    <h2
      className="text-left mt-2 mb-3"
    >
      Sign In
    </h2>
    <form
      className="m-0 mw-420"
    >
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginEmail"
        >
          Email
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginEmail"
          name="email"
          onChange={[Function]}
          placeholder="username@domain.com"
          type="email"
          value=""
        />
        <p
          className="mb-4"
        >
          The email address you used to register with edX.
        </p>
        <strong
          className="invalid-feedback"
          id="email-invalid-feedback"
        >
          The email address you've provided isn't formatted correctly.
        </strong>
      </div>
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginPassword"
        >
          Password
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginPassword"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <strong
          className="invalid-feedback"
          id="password-invalid-feedback"
        >
          Please enter your password.
        </strong>
      </div>
      <button
        className="mt-1 field-link"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
          data-icon="caret-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Need help signing in?
      </button>
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      >
        <div
          style={
            Object {
              "padding": ".1px 0",
            }
          }
        />
      </div>
      <a
        className="field-link mt-0 mb-3"
        href="http://localhost:18000/enterprise/login"
        onClick={[Function]}
        target="_self"
      >
        Sign in with your company or school
      </a>
      <button
        aria-disabled={true}
        aria-live="assertive"
        className="pgn__stateful-btn pgn__stateful-btn-state-pending btn-primary submit disabled btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <span
          className="d-flex align-items-center justify-content-center"
        >
          <span
            className="pgn__stateful-btn-icon"
          >
            <span
              aria-hidden={true}
              className="icon fa fa-spinner fa-spin"
              id="Icon1"
            />
          </span>
          Sign in
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`LoginPage should show error message 1`] = `
<div
  className="d-flex justify-content-center m-4"
>
  <div
    className="d-flex flex-column"
  >
    <div
      className="fade alert alert-danger show"
      role="alert"
    >
      <div
        className="alert-heading h4"
      >
        <span>
          We couldn't sign you in.
        </span>
      </div>
      <ul>
        <li>
          Email or password is incorrect.
        </li>
      </ul>
    </div>
    <div
      className="d-flex flex-row"
    >
      <p>
        First time here?
        <a
          className="ml-1"
          href="/register"
        >
          Create an Account
          .
        </a>
      </p>
    </div>
    <h2
      className="text-left mt-2 mb-3"
    >
      Sign In
    </h2>
    <form
      className="m-0 mw-420"
    >
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginEmail"
        >
          Email
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginEmail"
          name="email"
          onChange={[Function]}
          placeholder="username@domain.com"
          type="email"
          value=""
        />
        <p
          className="mb-4"
        >
          The email address you used to register with edX.
        </p>
        <strong
          className="invalid-feedback"
          id="email-invalid-feedback"
        >
          The email address you've provided isn't formatted correctly.
        </strong>
      </div>
      <div
        className="form-group mb-0 w-100"
      >
        <label
          className="h6 mr-1 form-label"
          htmlFor="loginPassword"
        >
          Password
        </label>
        <input
          aria-describedby=""
          className="form-control"
          id="loginPassword"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <strong
          className="invalid-feedback"
          id="password-invalid-feedback"
        >
          Please enter your password.
        </strong>
      </div>
      <button
        className="mt-1 field-link"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-caret-right fa-w-6 mr-1"
          data-icon="caret-right"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 192 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Need help signing in?
      </button>
      <div
        className="pgn-transition-replace-group position-relative"
        style={
          Object {
            "height": null,
          }
        }
      >
        <div
          style={
            Object {
              "padding": ".1px 0",
            }
          }
        />
      </div>
      <a
        className="field-link mt-0 mb-3"
        href="http://localhost:18000/enterprise/login"
        onClick={[Function]}
        target="_self"
      >
        Sign in with your company or school
      </a>
      <button
        aria-disabled={false}
        aria-live="assertive"
        className="pgn__stateful-btn pgn__stateful-btn-state-default btn-primary submit btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        <span
          className="d-flex align-items-center justify-content-center"
        >
          Sign in
        </span>
      </button>
    </form>
  </div>
</div>
`;
